lib/matobj2.gd

```diff
@@ -1813,3 +1816,339 @@ DeclareOperation( "Randomize", [ IsMatrixOrMatrixObj and IsMutable, IsRandomSour
+##
+##  <Description>
+##  <P/>
+##  Adds the product of <A>elm</A> with the <A>j</A>-th column of the mutable matrix <A>mat</A> to its <A>i</A>-th 
+##  column in-place. The <A>j</A>-th column is multiplied with <A>elm</A> from the left.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareOperation( "AddMatrixColumnsLeft", [ IsMatrixObj and IsMutable, IsInt, IsInt, IsObject ] );
+
+############################################################################
+##
+##  <#GAPDoc Label="SwapMatrixRows">
+##  <ManSection>
+##  <Oper Name="SwapMatrixRows" Arg='mat,i,j'/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  Swaps the <A>i</A>-th row and <A>j</A>-th row of a mutable matrix <A>mat</A>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareOperation( "SwapMatrixRows", [ IsMatrixObj and IsMutable, IsInt, IsInt ] );
+
+############################################################################
+##
+##  <#GAPDoc Label="SwapMatrixColumns">
+##  <ManSection>
+##  <Oper Name="SwapMatrixColumns" Arg='mat,i,j'/>
+##  
+##  <Returns>nothing</Returns>
+##
+##  <Description>
+##  <P/>
+##  Swaps the <A>i</A>-th column and <A>j</A>-th column of a mutable matrix <A>mat</A>.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareOperation( "SwapMatrixColumns", [ IsMatrixObj and IsMutable, IsInt, IsInt ] );
+
+
+#############################################################################
+##
+##  Open items:
+##
+##  - renaming issues:
+##    - There are functions 'RankMat', 'IsDiagonalMat', etc.
+##      We should consistently introduce the corresponding names
+##      'RankMatrix', 'IsDiagonalMatrix', etc. as main names,
+##      and keep the 'Mat' names as synonyms,
+##      and also keep these old names as documented if applicable.
+##
+##    - There are functions 'BaseIntMat', 'TriangulizeIntegerMat', etc.
+##      Introduce the corresponding names 'BaseIntegerMatrix',
+##      'TriangulizeIntegerMatrix', etc. as main names, keep the
+##      'IntMat' or 'IntegerMat' names as synonyms,
+##      and also keep the old names as documented if applicable.
+##
+##  - There is a problem with the operation 'Matrix'.
+##    An *attribute* with this name is declared for certain semigroups,
+##    in 'lib/reesmat.gd'.
+##    In 'lib/matobj2.gd', 'Matrix' is declared as a *mutable attribute*
+##    for lists; if it would not be declared as such then GAP's standard
+##    machinery would cause that 'Matrix( <list> )' returns an *immutable*
+##    value, which is not acceptable.
+##    The declaration as a mutable attribute in 'lib/matobj2.gd' works
+##    because this file is read before 'lib/reesmat.gd',
+##    but the consequence is that the attribute values in the semigroup
+##    context are also mutable (try the manual example for 'Matrix');
+##    this is a bug.
+##    (One could turn any attribute into a mutable one, by adding a
+##    corresponding declaration early enough.)
+##
```

Comment:
I think this text was added back by accident, and needs to be removed again.

Reply:
Done

Reply:
I'm sorry. I think that happened during a merge.